dozer-orchestrator/src/main.rs

```diff
@@ -31,6 +35,66 @@ fn render_logo() {
     info!("\nDozer Version: {VERSION}\n");
 }
 
+#[derive(Deserialize, Debug)]
+struct DozerPackage {
+    #[serde(rename(deserialize = "latestVersion"))]
+    pub latest_version: String,
+    #[serde(rename(deserialize = "availableAssets"))]
+    pub _available_assets: Vec<String>,
+    pub link: String,
+}
+
+fn version_to_vector(version: &str) -> Vec<i32> {
+    version.split('.').map(|s| s.parse().unwrap()).collect()
+}
+
+fn compare_versions(v1: Vec<i32>, v2: Vec<i32>) -> bool {
+    for i in 0..v1.len() {
+        match v1.get(i).cmp(&v2.get(i)) {
+            Ordering::Greater => return true,
+            Ordering::Less => return false,
+            Ordering::Equal => continue,
+        }
+    }
+    false
+}
+
+async fn check_update() {
+    use std::println as info;
+    const VERSION: &str = env!("CARGO_PKG_VERSION");
+    const ARCH: &str = std::env::consts::ARCH;
+    const OS: &str = std::env::consts::OS;
+
+    loop {
+        info!("Checking for updates...");
+        let request_url = format!(
+            "https://metadata.dev.getdozer.io/?version={}&build={}&os={}",
+            VERSION, ARCH, OS
+        );
+        let response = reqwest::get(&request_url).await;
+        match response {
+            Ok(r) => {
+                let package: DozerPackage = r.json().await.unwrap();
+                let current = version_to_vector(VERSION);
+                let remote = version_to_vector(&package.latest_version);
+
+                if compare_versions(remote, current) {
```

Comment:
Is it printing this message on every iteration? We should only print on first I believe

Reply:
We print and stop checking?

Reply:
makes sense